server/src/main/java/org/opensearch/search/pipeline/TrackingSearchRequestProcessorWrapper.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.core.action.ActionListener;
+
+/**
+ * Wrapper for SearchRequestProcessor to track execution details.
+ *
+ *  @opensearch.internal
+ */
+public class TrackingSearchRequestProcessorWrapper implements SearchRequestProcessor {
+
+    private final SearchRequestProcessor wrappedProcessor;
+
+    /**
+     * Constructor for the wrapper.
+     *
+     * @param wrappedProcessor the actual processor to be wrapped
+     */
+    public TrackingSearchRequestProcessorWrapper(SearchRequestProcessor wrappedProcessor) {
+        this.wrappedProcessor = wrappedProcessor;
+    }
+
+    @Override
+    public String getType() {
+        return wrappedProcessor.getType();
+    }
+
+    @Override
+    public String getTag() {
+        return wrappedProcessor.getTag();
+    }
+
+    @Override
+    public String getDescription() {
+        return wrappedProcessor.getDescription();
+    }
+
+    @Override
+    public boolean isIgnoreFailure() {
+        return wrappedProcessor.isIgnoreFailure();
+    }
+
+    @Override
+    public SearchRequest processRequest(SearchRequest request) throws Exception {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SearchRequest processRequest(SearchRequest request, PipelineProcessingContext requestContext) throws Exception {
+        try {
+            return wrappedProcessor.processRequest(request, requestContext);
+        } catch (UnsupportedOperationException e) {
+            throw e;
+        }
+    }
```

Comment:
Oh -- we were talking over Slack about this one, but actually I meant that you could throw an `UnsupportedOperationException` from this method.

The only `process*` method that needs to be implemented is `processRequestAsync`. That's the one that gets called from `Pipeline`.